src/main/java/mate/academy/spring/validator/EmailValidator.java

```diff
@@ -0,0 +1,18 @@
+package mate.academy.spring.validator;
+
+import java.util.regex.Pattern;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class EmailValidator implements
+        ConstraintValidator<Email, String> {
+    private static final Pattern EMAIL_REGEX =
+            Pattern.compile("^(?=.{1,64}@)[A-Za-z0-9_-]"
+            + "+(\\\\.[A-Za-z0-9_-]+)*@[^-][A-Za-z0-9-]"
+            + "+(\\\\.[A-Za-z0-9-]+)*(\\\\.[A-Za-z]{2,})$");
+
+    @Override
+    public boolean isValid(String email, ConstraintValidatorContext constraintValidatorContext) {
+        return email != null && email.matches(EMAIL_REGEX.pattern());
```

Comment:
Under hood the method `matches` creates an object of a type Pattern compiling the string you pass. Let's do not use the method `matches` if you already have a `Pattern` object